src/lib/github/addIssuesToRepo.ts

```diff
@@ -11,21 +12,35 @@ export default async function addIssuesToRepo (context: Context, newIssues: Issu
   const octokit = context.github;
   const { owner, repo } = getBasicRepoProps (context);
   const ghIssues = await getAllRepoIssues(context);
+
   newIssues.forEach(async ({title, body, authors}) => {
     const assignees = uniq(authors);
-    const currentGHIssue = find(ghIssue => ghIssue.title === title, ghIssues);
+
+    /**
+     * Am up to writing a function that removes checked out or no longer existing issue comments.
+     */
+
+    const currentGHIssueComments = find(ghIssueComment => ghIssueComment.comments === body, ghIssues);
+    const newGHIssueComments = currentGHIssueComments.comments.includes(" - [x] ") || !(currentGHIssueComments.comments === body) ? replaceCommentArray(currentGHIssueComments.comments, body) : currentGHIssueComments.comments;
+
     const fields = {
       owner,
       repo,
       body,
       title,
       labels: ["GH-TODO-BOT"],
       assignees,
-      ...(currentGHIssue ? { number: currentGHIssue.number } : {})
+      ...(currentGHIssueComments ?  {number: currentGHIssueComments.number, comments: newGHIssueComments } : {})
```

Comment:
The fields variable contains arguments passed to github API functions for either adding a new issue or updating a new issue with a new set of comments (body). The comments are already within the `body` field.

Both those API functions don't take in a comments field. 
See: 
https://developer.github.com/v3/issues/#edit-an-issue
https://developer.github.com/v3/issues/#create-an-issue

Hence there is no need for changing this line of code. It should remain as is.

Reply:
So true